eng/versioning/set_versions.py

```diff
@@ -93,23 +109,133 @@ def prep_version_file_for_source_testing(build_type):
+                    vmatch = version_regex_named.match(module.current)
+                    temp_ver = '{}.{}.{}'.format(vmatch.group('major'), vmatch.group('minor'), vmatch.group('patch'))
+                    # we should never have buildmetadata in our versioning scheme
+                    if vmatch.group('buildmetadata') is not None:
+                        raise ValueError('artifact ({}) version ({}) in version file ({}) is not a correct version to release. buildmetadata is set and should never be {}'.format(artifact_id, module.current, version_file, vmatch.group('buildmetadata'))) 
+
+                    # reconstruct the version from the semver pieces and it should match exactly the current
+                    # version in the module
+                    # If there's a pre-release version it should be beta.X
+                    if vmatch.group('prerelease') is not None:
+                        prerel = vmatch.group('prerelease')
+
+                        if prerelease_regex_named.match(prerel) is None:
+                            raise ValueError('artifact ({}) version ({}) in version file ({}) is not a correct version to release. The accepted prerelease tag is (beta.X) and the current prerelease tag is ({})'.format(artifact_id, module.current, version_file, prerel))
+
+                        prever = prerelease_regex_named.match(prerel)
+                        rev = int(prever.group('revision'))
+                        temp_ver = '{}-beta.{}'.format(temp_ver, str(rev))
+
+                    # last but not least, for sanity verify that the version constructed from the
+                    # semver pieces matches module's current version
+                    if module.current != temp_ver:
+                        raise ValueError('artifact ({}) version ({}) in version file ({}) does not match the version constructed from the semver pieces ({})'.format(artifact_id, module.current, version_file, temp_ver)) 
+
+                    print('The version {} for artifact_id {} looks good!'.format(module.name, module.current))
+                    
+
+    if not artifact_found:
+       raise ValueError('artifact_id ({}) was not found in version file {}'.format(artifact_id, version_file))
+
 def main():
     parser = argparse.ArgumentParser(description='set version numbers in the appropriate version text file')
     parser.add_argument('--update-type', '--ut', type=UpdateType, choices=list(UpdateType))
     parser.add_argument('--build-type', '--bt', type=BuildType, choices=list(BuildType))
     parser.add_argument('--build-qualifier', '--bq', help='build qualifier to append onto the version string.')
     parser.add_argument('--artifact-id', '--ar', help='artifactId to target.')
     parser.add_argument('--prep-source-testing', '--pst', action='store_true', help='prep the version file for source testing')
+    parser.add_argument('--increase-version', '--iv', action='store_true', help='increase the version for a given artifact')
```

Comment:
nit: minor but the other scripts call this increment-version.

Reply:
changed to increment, will be updated with next commit